### ISSUE TITLE
Support for List<T> on Converters

### ISSUE THREAD

**Issue by brunobat** [smallrye/smallrye-converters#9]
This is a suggestion to add support to convert List<T> of objects to the Converters API interface. 
Probably it should be Collection<T>, but lists are simpler for a first version.

**Comment by dmlloyd** [comment]
Are you thinking as a general API enhancement? Because support exists in the implementation class `io.smallrye.converters.Converters#newCollectionConverter`.

**Comment by brunobat** [comment]
Yes, as an enhancement. Lists are everywhere and supporting them out of the box would be handy.
